godot-core/src/builtin/callable.rs

```diff
@@ -0,0 +1,178 @@
+            })
+        }
+    }
+
+    /// Calls the method represented by this callable. Arguments passed should match the method's
+    /// signature. If called with more arguments than expected by the method, the extra arguments will be
+    /// ignored. If called with fewer arguments than expected it will crash. If called with arguments of the
+    /// wrong type then an error will be printed and the call fails.
+    ///
+    /// _Godot equivalent: `callv`_
+    pub fn callv(&self, arguments: VariantArray) -> Variant {
+        self.as_inner().callv(arguments)
+    }
+
+    /// Returns the name of the method represented by this callable. If the callable is a lambda function,
+    /// returns the function's name.
+    ///
+    /// _Godot equivalent: `get_method`_
+    pub fn method_name(&self) -> Option<StringName> {
+        let method_name = self.as_inner().get_method();
+        if method_name.is_empty() {
+            None
+        } else {
+            Some(method_name)
+        }
+    }
+
+    /// Returns the object on which this callable is called.
+    ///
+    /// _Godot equivalent: `get_object`_
+    pub fn object(&self) -> Option<Gd<Object>> {
+        // Increment refcount because we're getting a reference, and `InnerCallable::get_object` doesn't
+        // increment the refcount.
+        self.as_inner().get_object().map(|object| {
+            <<Object as GodotClass>::Mem as crate::obj::mem::Memory>::maybe_inc_ref(&object);
+            object
+        })
+    }
+
+    /// Returns the ID of this callable's object, see also [`Gd::instance_id`].
+    ///
+    /// _Godot equivalent: `get_object_id`_
+    pub fn object_id(&self) -> Option<InstanceId> {
+        let id = self.as_inner().get_object_id();
+        InstanceId::try_from_i64(id)
+    }
+
+    /// Returns the 32-bit hash value of this callable's object.
+    ///
+    /// _Godot equivalent: `hash`_
+    pub fn hash(&self) -> u32 {
+        self.as_inner().hash().try_into().unwrap()
+    }
+
+    /// Returns true if this callable is a custom callable. Custom callables are mainly created from bind or
+    /// unbind. In GDScript, lambda functions are also custom callables.
+    ///
+    /// _Godot equivalent: `is_custom`_
+    pub fn is_custom(&self) -> bool {
+        self.as_inner().is_custom()
+    }
+
+    /// Returns true if this callable has no target to call the method on. This is not the negated form of
+    /// [`is_valid`], as `is_valid` will return `false` if the callable has a target but the method does not
+    /// exist.
+    ///
+    /// _Godot equivalent: `is_null`_
+    pub fn is_null(&self) -> bool {
+        self.as_inner().is_null()
+    }
+
+    /// Returns true if this callable is a standard callable. This method is the opposite of [`is_custom`].
+    /// Returns `false` if this callable is a lambda function.
+    ///
+    /// _Godot equivalent: `is_standard`_
+    pub fn is_standard(&self) -> bool {
+        self.as_inner().is_standard()
+    }
```

Comment:
Should we remove either `is_standard`/`is_custom`, so that the provided getter methods are actually orthogonal to each other?

Reply:
Im not sure, i feel like a Callable if implemented in rust would effectively be a kind of enum, something like
```rs
enum Callable {
  Standard(Gd<Object>, StringName),
  Custom(Box<dyn CustomCallable>)
}
```
in which case it is fairly common to have an `is_x` for every case.

Reply:
Yeah, but at the moment it's more like
```rs
enum Callable {
  Standard(Gd<Object>, StringName),
  Invalid(Gd<Object>, StringName),     // method does not exist
  Custom(Box<dyn CustomCallable>),
  Null,
}
```
But even that is not accurate, because `standard == !custom` according to Godot docs.

It's just that the mental model is already quite complex with multiple states, and having redundant methods doesn't exactly help.

Reply:
i guess we could remove `is_standard` then, since callables are generally assumed to be a standard callable, and custom callables are the exception.

Reply:
Sounds good to me!
Maybe we can still mention the relation of standard/custom inside `is_custom`, or provide a doc alias...

Reply:
we should probably provide doc aliases for all functions that diverge from the godot name tbh, but yeah a doc alias in this case at least.